include/ignition/math/Plane.hh

```diff
@@ -119,6 +122,76 @@ namespace ignition
         return this->normal.Dot(_point) - this->d;
       }
 
+      /// \brief Given x,y point find corresponding Z point on plane.
+      /// \param[in] x - 2d x coordinate.
+      /// \param[in] y - 2d y coordinate.
+      /// \param[in] eps - The value to consider zero. Defaults to 1e-16
+      /// \return coincident point.
+      public: Vector3<T> GetPointOnPlane(const T x,
+        const T y,
+        const T eps = 1e-16) const
+      {
+        auto z_val = (std::fabs(this->Normal().Z()) > eps) ?
+         (this->Offset() - (this->Normal().Dot({x, y, 0})))/this->Normal().Z()
+         : 0;
+        auto coincidentPoint = Vector3<T>{x, y, z_val};
+        return coincidentPoint;
+      }
+
+      /// \brief Get the intersection of a line with the plane
+      /// given the line's gradient and a point in parametrized space.
+      /// \param[in] _point A point that lies on a line.
+      /// \param[in] _gradient The gradient of the line.
+      /// \return The point of intersection. std::nullopt if the line is
+      /// parrallel to the plane.
+      public: std::optional<Vector3<T>> Intersect(
+        const Vector3<T> &_point,
+        const Vector3<T> &_gradient) const
```

Comment:
Can we use `Line3` instead of a point + gradient? That should be more well defined.

Reply:
`Line3`s have a start and an end the `Intersect()` routine would look different from the point of view of a `Line3` vs an infinite line.